File: DayFour/Python/transaction_account/transaction_recorder.py
class TransactionRecorder:
    depositRecord = []
    logapp = 0


    def __init__(self,logapp):
        self.logapp = logapp

        

    def depositRecorder(self,amount):
        newBalance = self.logapp.getBalance();
        self.depositRecord.append(f"Deposited: {amount} + newBalance is {newBalance}");    



    def withdrawRecorder(self,amount):
        newBalance = self.logapp.getBalance();
        self.depositRecord.append(f"Withdrawal:  {amount} + newBalance is {newBalance}");    


    def getDepositHistory(self):

        return self.depositRecord

File: DayFour/Java/TransactionLogApp/TransactionRecorder.java
import java.util.ArrayList;

public class TransactionRecorder{
    private ArrayList<String> depositRecord = new ArrayList<>();
    private TransactionLogApplication logapp;


    public TransactionRecorder(TransactionLogApplication logapp){
        this.logapp = logapp;

    }

public void depositRecorder(double amount){
    double newBalance = this.logapp.getBalance();
    this.depositRecord.add("Deposited: " + amount + "newBalance is" + newBalance);    

}

public void withdrawRecorder(double amount){
    double newBalance = this.logapp.getBalance();
    this.depositRecord.add("Withdrawal: " + amount + "newBalance is" + newBalance);    

}

public ArrayList<String> getDepositHistory(){

    return this.depositRecord;
}


}
